### DISCUSSION TITLE
Issues using a different VESC

### DISCUSSION THREAD

**Discussion by alimohammadhadi** [prl-mushr/mushr#68] [Hardware]
![95852982329009514](https://user-images.githubusercontent.com/38579308/147315904-c490fe83-4789-4da5-af4c-89a2d901843c.png)

Hello everyone. 
I am using a VESC 75/300 with 5.2 firmware preinstalled. I used windows based VESC tools and easily tuned the PID for the motor. The Windows PC was successfully able to run the motor. The firmware was the default file and there is no custom file with "servoout.bin". 

However, now that I am in Linux, I can't run BLDC tools because of the version mismatch. I thought that it wouldn't be a problem because I have already configured the VESC in windows but I get this error:
![image](https://user-images.githubusercontent.com/38579308/147316288-5b8300ce-feb6-4252-8409-ecf1d0011928.png)
terminate called after throwing an instance of 'serial::SerialException'
what():  SerialException device reports readiness to write but returned no data

What I do not understand is that BLDC Tools GitHub provides the file "tool/firmwares/hw_410_411_412/VESC_servoout.bin." which may not be compatible with my VESC. Should I somehow force this file to my VESC even if it can potentially damage it?

Does anyone know how to resolve these problems that arise due to different VESC or hardware?

**Comment by palanc** [comment]
Hi @alimohammadhadi - unfortunately we only support VESCs with HW version 4.12 and don't have any experience with using VESC 75/300. I have looked around a little bit and haven't been able to find a 'servoout.bin' file that matches your hardware version. In fact, does the VESC 75/300 even have a physical output for controlling a servo motor?  I agree that it would be very risky to upload the HW version 4.12 'servoout.bin' to your VESC. The only advice I can give is to instead buy the VESC listed in our bill of materials, sorry that I don't have a better answer.

**Comment by precioso** [comment]
Hello,

Do you know if a VESC 4.20 would make it? It is very difficult to find at a reasonable price the 4.12 version but we can more easily find the 4.2 one.

Thank you for your answer.

Best regards,

**Reply by palanc** [reply]
Hi @precioso - Sorry for the late reply.

If you are referring to either of the VESCs below, I would expect them to work since they are based on 4.12 and the changes they described shouldn't prevent them from being compatible as far as I can tell. However, we have not actually tested this ourselves so do not know for sure (@naughtyStark please correct me if I'm wrong).

[https://flipsky.net/products/mini-fsesc4-20-50a-base-on-vesc-widely-used-in-eskateboard-escooter-ebike](https://flipsky.net/products/mini-fsesc4-20-50a-base-on-vesc-widely-used-in-eskateboard-escooter-ebike)
[https://www.amazon.com/Flipsky-Electric-Controller-Skateboard-FSESC4-20/dp/B08725X8CT](https://www.amazon.com/Flipsky-Electric-Controller-Skateboard-FSESC4-20/dp/B08725X8CT)

Note however that the VESCs I linked above seem to come with heat sinks, which you may need to remove in order to make them fit onto the car.

Let us know if you have any other questions!

**Reply by sidtalia** [reply]
@precioso We haven't tested with the hardware specified yet. As @palanc stated, it "should" work since it's the same firmware version,  but it is hard to know for sure. 
@precioso the VESC 4.2 seems to be just as expensive as a 4.12 here:https://flipsky.net/products/torque-esc-vesc-%C2%AE-bldc-electronic-speed-controller
Is there a different reason that makes it difficult to find the 4.12 at a reasonable price?

Regards,
Sidharth.
